### DISCUSSION TITLE
Trouble using OTG on stm32f105

### DISCUSSION THREAD

**Discussion by paulmrr** [hathach/tinyusb#1116] [Q&A] [answered]
I am having some trouble porting an example to an stm32f105. I have been trying to go based off of one of the stm32f4 examples since it also uses the OTG peripheral. I commented out some of incompatible code (led and gpio stuff) in the family and board.c files. I also specified in the Makefile that I was using an F1 device instead of F4. When I debug the USB_FS interrupt gets hit once in the beginning, but never again. My computer does not detect the USB.
I noticed that there was a header (dcd_common.h) file in the dcd_synopsys directory that redefined some of the registers to make it work with cortex M3. I included this file, but it did not solve my problem. Any advice on how to narrow down the problem? Thanks!

**Comment by hathach** [comment, marked as answer]
to make it easier for others to follow and help with your issue, you should provide the **buildable** links to your github fork, where  you add your f105 board settin. This helps to make sure others can reproduce the issue and/or spot any problem in your code.

**Reply by paulmrr** [reply]
Hi Ha, Thank you for your quick response. I was able to get Tiny USB to work with my board when I tested the Tiny USB code on its own (without my own code). In my project, I had moved a lot of the tiny usb files into my project's Source directory and somewhere in the mix there must have been some dependency missing. I have still not figure out what the problem was, but I will keep looking at it.

**Comment by duempel** [comment]
Hello Paul, as hathach already said it would be good to get a view on your source.
Since you are using a custom hardware (I think so because I do not know any F105 boards with USB connector) it would be good to know if your hardware is working correctly. Maybe you could use ST's USB stack to check if everything is good. Problems could be using a wrong oscillator or using other peripherals on PA9 pin (this should be VBUS detection).
After excluding hardware issues I could imagine that a define is missing. Did you also set STM32F105x8, STM32F105xB or STM32F105xC? TinyUSB checks this define to see if the F1 MCU uses ST's FS or Synopsis OTG peripheral. Also make sure the clock settings are matching your oscillator.

**Reply by paulmrr** [reply]
Thanks Jan for your suggestions! I was able to get the project to work, by just focusing on getting the Tiny USB code to work independent of my own project's code. I believe I had all the proper defines, but I think there was either an issue with the clock initialization or something else. One potential problem?- I included HAL files in my own project and then I also was using HAL files from the Tiny USB directory, so it might have been the case that I thought I was using one HAL header instead of the other- stm32f1xx_hal_conf.h for example. I was using my own Makefile and had include paths for TinyUSB's HAL directory (-Itinyusb/hw/mcu/st ...etc.) and my own project's HAL library (-IDrivers/stm32f1xx_hal/...etc.).
